openssl/src/crypto/pkey.rs

```diff
@@ -69,6 +70,22 @@ impl PKey {
         }
     }
 
+    /// Reads private key from PEM, takes ownership of handle
```

Comment:
Why would the struct representing a private key have a function to read a public key?